lexers/e/ec.go

```diff
@@ -0,0 +1,36 @@
+package e
+
+import (
+	"regexp"
+
+	. "github.com/alecthomas/chroma" // nolint
+	"github.com/alecthomas/chroma/lexers/internal"
+)
+
+var (
+	ecAnalyserIncludeRe = regexp.MustCompile(`(?m)^\s*#include [<"]`)
+	ecAnalyserIfdefRe   = regexp.MustCompile(`(?m)^\s*#ifn?def `)
+)
+
+// Ec lexer.
+var Ec = internal.Register(MustNewLexer(
+	&Config{
+		Name:      "eC",
+		Aliases:   []string{"ec"},
+		Filenames: []string{"*.ec", "*.eh"},
+		MimeTypes: []string{"text/x-echdr", "text/x-ecsrc"},
+	},
+	Rules{
+		"root": {},
+	},
+).SetAnalyser(func(text string) float32 {
```

Comment:
This is also the text analysis from C lexer, right? If so, let's do it like we did for Cuda lexer: https://github.com/wakatime/chroma/pull/82

Reply:
yes for sure. tks.